pkg/agent/config.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2019 Red Hat and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Defines Nimbess agent configuration.
+package agent
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const (
+	L2DriverMode = "layer2"
+	BESS = "BESS"
+)
+
+// INI config types
+type Network struct {
+	Driver		string	`mapstructure:"driver"`
+	MacLearn	bool	`mapstructure:"mac_learning"`
+	TunnelMode	bool	`mapstructure:"tunnel_mode"`
+	FIBSize		int64	`mapstructure:"fib_size"`
+}
+
+type NimbessConfig struct {
```

Comment:
golint I think will ask you to add comments to your publicly exported types and method in the format of:

```
// NimbessConfig $is_however_you_want_to_describe_it_here
```
I personally like to keep in that mindframe in case you ever wanna gen docs off your code, but, it doesn't "actually matter, but, fwiw

Reply:
Thanks. There's a lot of golint issues. I'm going to setup travis and add it to this commit as well so we can see it pass at least golint